File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equipable/CobaltShield.java
package org.rexcellentgames.burningknight.entity.item.accessory.equipable;

import org.rexcellentgames.burningknight.assets.Locale;

public class CobaltShield extends Equipable {
	{
		name = Locale.get("cobalt_shield");
		description = Locale.get("cobalt_shield_desc");
		sprite = "item-obsidian_shield";
	}

	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);
		this.owner.modifyStat("knockback", -1);
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);
		this.owner.modifyStat("knockback", 1);
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/CobaltShield.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

import org.rexcellentgames.burningknight.assets.Locale

class CobaltShield : Equippable() {
	init {
		name = Locale.get("cobalt_shield")
		description = Locale.get("cobalt_shield_desc")
		sprite = "item-obsidian_shield"
	}

	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.modifyStat("knockback", -1f)
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.modifyStat("knockback", 1f)
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}
}